boards/posix/nrf52_bsim/Kconfig.defconfig

```diff
@@ -18,6 +18,20 @@ config BT_CTLR
 	default y
 	depends on BT
 
+config UART_NATIVE_POSIX
```

Comment:
This is even worse.. the native_posix console is not meant for the nrf52_bsim. It can be made to work but is very hacky. Setting it to be default in the nrf52_bsim is asking for a lot of trouble.
If you need it in some app, you can always set it for that app, in the worst case as an overlay for the nrf52_bsim board

Reply:
@aescolar . I'm wondering if there is some nice way to get the console subsystem to work nicely with both native_posix and nrf52_bsim. I don't see why we should need to differentiate between those. It would be nice if those could share as much as possible

Reply:
@rugeGerritsen There is a nice way of doing it:
For the nrf52_bsim implementing a model of the UART HW, and enabling the real nRF UART driver.
native_posix and the nrf52_bsim only share the POSIX arch (and the inf SOC which provides CPU boot like functionality), but otherwise are not meant to share anything because they are built with different purposes in quite different ways.
The direction you are proposing is merging both into an amalgam. But that would not be able to fulfill the original purpose of either.